### PR TITLE
added delete_entry, add_entry + readme cleanup

### PR DESCRIPTION
**Pull request by Evidlo**
- added `delete_entry`, `delete_group` functions
- cleaned up the readme and added examples for deleting entries and groups
- fixed exception in `find_entries_by_path` if parent group is not found

### CODE REVIEW

File: pykeepass/pykeepass.py [hunk 1]
```diff
@@ -194,11 +194,14 @@ def find_entries_by_path(self, path, regex=False, tree=None, history=False, firs
...
         group_path = os.path.dirname(path)
         group = self.find_groups_by_path(group_path, tree=tree, regex=regex, first=True)
+
         if group is not None:
             if regex:
                 res = [x for x in group.entries if re.match(entry_title, x.title)]
             else:
                 res = [x for x in group.entries if x.title == entry_title]
+        else:
+            return None
```
Comment (pschmitt) [inline pykeepass/pykeepass.py]: None is implied when using return w/o any argument.

File: .gitignore [hunk 2]
```diff
@@ -3,3 +3,4 @@
 dist/
 *.kdbx
 *.egg-info/
+**/TAGS
```
Comment (pschmitt) [inline .gitignore]: What's that?

File: README.rst [hunk 3]
```diff
@@ -34,6 +34,7 @@ Simple Example
 
    # create a new entry
    >>> kp.add_entry(group, 'gmail', 'myusername', 'myPassw0rdXX')
+   [Entry: "email/gmail"]
```
Comment (pschmitt) [inline README.rst]: Isn't that supposed to return the entry, and not a list?

**Comment by Evidlo** [comment]
I have no idea how to correctly rebase this.  I keep making the history worse.

**Comment by pschmitt** [comment]
To rebase, create a new branch and cherry pick the commits. To facilitate future reviews it'd be better to open a PR with a specific changes, and not a big one with all changes.

https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging
https://git-scm.com/docs/git-cherry-pick

**Pull request closed without merging**